frontend/src/js/data_compilers.js

```diff
@@ -0,0 +1,169 @@
+     This value is used when comparing two departures in `compareDepartures`.
+     Returns: float: Amount of minutes left until departure according to current time.
+    */
+    var upd = new Date(latest_updated)
+    let minutes_left
+    if (displaytime == "Nu"){
+        minutes_left = 0
+    }
+    else if(displaytime == "-"){
+        // Hög prioritet men störning i trafiken
+        return 0 // real_minutes_left
+    }
+    else if (displaytime.includes(":")){
+        let time = displaytime.split(":")
+        let hours = time[0]
+        let mins = time[1]
+        let depart = new Date(latest_updated)
+        depart.setHours(hours)
+        depart.setMinutes(mins)
+
+        // departure can't be before latest_updated, this handles edge cases at midnight.
+        if ((depart - upd) <= 0){
+            // increment day by 1
+            depart.setDate(depart.getDate() + 1)
+        }
+        minutes_left = (depart - upd)/1000/60
+    }
+    else {
+        minutes_left = parseInt(displaytime.split(" ")[0], 10)
+    }
+    let real_departure = minutes_left*60 + upd/1000 //unix seconds
+    let real_minutes_left = (real_departure - now/1000)/60
+    return real_minutes_left // float
+};
+
+var compareDepartures = function (a, b){
+    //Compares two departures, used for calling sort()
+    return a.minutes_left - b.minutes_left
+};
+
+var getDisplayTimeText = function (ride){
+    if (ride.ExpectedDateTime && ride.minutes_left > 25){
+        // return time in HH:MM format
+        return new Date(ride.ExpectedDateTime).toLocaleTimeString().substr(0,5)
+    }
+    else if (ride.DisplayTime == "-"){
+        return "Okänt"
+    }
+    else {
+        // Minutes left from now, could do some converting to HH:MM format if it's too long.
+        return Math.round(ride.minutes_left) + " min"
+    }
+};
+
+// ######   Functions that compile the data   ######
+
+var compileRides = function (stations){
+    /*
+     Returns all rides in stations as a single list of objects with some modified attributes
+     such as `minutes_left`, `type` and a proper `DisplayTime`.
+    */
+
+    let rides = []
+    let types = ["Metros", "Trams", "Buses"]
+
+    let now = new Date()
+    stations.forEach(station =>{
+        types.forEach(type => {
+            if (station.Departures[type]){
+                station.Departures[type].forEach(ride => {
+                    ride.minutes_left = realMinutesLeft(ride.DisplayTime, station.Departures.LatestUpdate, now)
+                    ride.DisplayTime = getDisplayTimeText(ride)
+                    ride.TransportMode = ride.TransportMode.toLowerCase()
+                    rides.push(ride)
+                })
+            }
+        })
+    })
+
+    // Keys are StopAreaNumbers (identifiers for stations), values are minutes_left required to be shown on screen.
+    let thresholds = {10194:5, 60080:5, 6601:5, 0:5, 10036:2}

```

Comment:
This list should possibly be declared at an earlier point, but it will suffice to simply note this as an issue and possibly fix it at a later point.

Reply:
The object thresholds? I tend to declare variables in proximity to where it is first used and preferably not before code where it isn't needed. I guess I could declare it earlier though.

I have converted it to a parameter in order to unit test it in a coming commit.

Reply:
You do have a point, as I cannot remember it being used at any other point. My argument is that if we need to update any of this information at a later point (for instance due to new stations being introduced), it would be smart to have them a place where it is easy to find them. We should at least note that this is the place to change things in such a case (as well as the to-be-commited python script).